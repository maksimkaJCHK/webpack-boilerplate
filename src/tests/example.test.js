import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import Example from './example_components/example.js';

test('Для примера', () => {
  expect(2 + 2).toBe(4);
});

describe('Example testing', () => {
  it('Первый тест', () => {
    const component = renderer.create(<Example />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Второй тест', () => {
    const example = shallow(<Example />);
    const itm = example.find('#disabled');
    expect(itm.length).toBe(1);
  });

  it('Третий тест', () => {
    const example = shallow(<Example />);
    const itm = example.find('#disabled');
    expect(itm.contains(<div id="disabled">Для примера</div>))
      .toBe(true);
  });

  it('Четвертый тест', () => {
    const example = shallow(<Example />);
    expect(example.find('#disabled').length)
      .toBe(1);
  });

});
