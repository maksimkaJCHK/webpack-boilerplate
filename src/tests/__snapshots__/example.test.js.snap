// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example testing Первый тест 1`] = `
<div
  id="example"
>
  <div
    id="disabled"
  >
    Для примера
  </div>
</div>
`;
